src/storage/file_io.c

```diff
@@ -2091,6 +2094,60 @@ fileio_close (int vol_fd)
     }
 }
 
+/*
+ * fileio_cache_dump () - Iterate through all the cached volumes
+ * and dump all the important information about the volume.
+ *    return: void
+ *    thread_p(in) - Thread entry
+ */
+void
+fileio_cache_dump (THREAD_ENTRY * thread_p)
+{
+  APPLY_ARG arg = { 0 };
+  (void) fileio_traverse_system_volume (thread_p, fileio_dump_sysvolinfo, &arg);
+  (void) fileio_traverse_permanent_volume (thread_p, fileio_dump_volinfo, &arg);
+  (void) fileio_traverse_temporary_volume (thread_p, fileio_dump_volinfo, &arg);
+}
+
+/*
+ * fileio_dump_cached_file () - Dump information about given volume.
+ *    return: void
+ *    volid(in):  Volume ID
+ *    vdes(in):   Volume descriptor
+ *    vlabel(in): Volume label
+ */
+static void
+fileio_dump_cached_file (VOLID volid, int vdes, const char *vlabel)
+{
+  _er_log_debug (ARG_FILE_LINE, "Cached volume with volid=%d, vdes=%d, label=%s\n", volid, vdes, vlabel);
+}
+
+/*
+ * fileio_dump_volinfo () - Iterator function to call fileio_dump_volinfo
+ * to print volume information.
+ *    return: false to keep the iteration going
+ *    vdvol_info_p(in): Volume information structure
+ */
+static bool
+fileio_dump_volinfo (THREAD_ENTRY *, FILEIO_VOLUME_INFO * vol_info_p, APPLY_ARG *)
+{
+  fileio_dump_cached_file (vol_info_p->volid, vol_info_p->vdes, vol_info_p->vlabel);
+  return false;
+}
+
+/*
+ * fileio_dump_sysvolinfo () - Iterator function to call fileio_dump_volinfo
+ * to print system volume information.
+ *    return: false to keep the iteration going
+ *    sys_volinfo(in): System volume information structure
+ */
+static bool
+fileio_dump_sysvolinfo (THREAD_ENTRY *, FILEIO_SYSTEM_VOLUME_INFO * sys_volinfo, APPLY_ARG *)
```

Comment:
I checked and the last param could be changed to `const APPLY_ARG *` as nothing modifies it. But maybe this is out of scope as it's a crosscutting change.

Reply:
I think is better to leave it as nonconst for now as it will fit better with the template of the other apply functions.

Reply:
I think @cristiarg suggestion is to update all the apply functions. But this is indeed very out of scope :).